File: frontend/src/types/playlist.ts
export interface PlaylistSummary {
  id: string
  name: string
  description: string
  cover_url: string | null
  song_count: number
  created_at: string
  updated_at: string
}

export interface PlaylistSong {
  id: string
  netease_song_id: number
  song_name: string
  artist_name: string
  cover_url: string
  added_at: string
}

export interface PlaylistDetail extends PlaylistSummary {
  songs: PlaylistSong[]
}

export interface CreatePlaylistRequest {
  name: string
  description: string
}

export interface AddSongToPlaylistRequest {
  netease_song_id: number
  song_name: string
  artist_name: string
  cover_url: string
}

File: backend/src/api/schemas/playlist.py
from pydantic import BaseModel, Field


class CreatePlaylistRequest(BaseModel):
    name: str = Field(min_length=1, max_length=50)
    description: str = Field(default="", max_length=200)


class UpdatePlaylistRequest(BaseModel):
    name: str = Field(min_length=1, max_length=50)
    description: str = Field(default="", max_length=200)


class AddSongRequest(BaseModel):
    netease_song_id: int
    song_name: str = Field(min_length=1, max_length=255)
    artist_name: str = Field(min_length=1, max_length=255)
    cover_url: str = Field(default="", max_length=512)
